src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -745,6 +745,15 @@ IEnumerator IEnumerable.GetEnumerator()
             return new Enumerator(this, Enumerator.KeyValuePair);
         }
 
+        int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
```

Comment:
also check capacity >= count. 

also return immediately if capacity <= existing capacity (ie entries.Length) since no work to do

Reply:
then you can keep the existing assert at the top of Resize.

Reply:
since I would immediately return when 
``` capacity <= existing capacity ```
and we know that
``` existing capacity >= count ```
Then we should not need to have a check for 
``` capacity >= count ```